ApeWorX/ape
# test: ability to run subprocess tests

### What I did

* Created a way to use subprocess to invoke ape commands rather than click runner, when appropriate.
* Migrated `ape plugins` tests to use the subprocess method (fixes: #285)
* Migrated `ape test` tests to use the subprocess method. **NOTE: I will need this to test some of the changes I am making to the gas reporting feature**.

### How I did it

Create a class with similar methods to the click runner but uses `subprocess.run` instead. `subprocess.run` calls out to the OS more and gets a more accurate read of installed packages. Also, it ensures more isolation from the already-running `pytest` instance.

### How to verify it

Try installing a plugin with click runner and checking the list output and seeing if it was installed. It never shows it as installed but then check yourself via a shell cmd manually and you will see it did install. Try the same thing with `subprocess.run()`, like this PR, and notice that you are able to see installation changes.

### Checklist
